sql/src/main/java/io/crate/analyze/Relations.java

```diff
@@ -31,16 +31,45 @@
 import io.crate.analyze.relations.TableFunctionRelation;
 import io.crate.analyze.relations.TableRelation;
 import io.crate.analyze.relations.UnionSelect;
+import io.crate.expression.symbol.Field;
 import io.crate.expression.symbol.Symbol;
 import io.crate.expression.symbol.Symbols;
 import io.crate.metadata.ColumnIdent;
 
 import java.util.Collection;
 import java.util.List;
 import java.util.function.Consumer;
+import java.util.function.Function;
 
 public class Relations {
 
+    // In a query like `select o['id'] from (select obj as o ..)`
+    // `o['id']` is not found in the child (its called `obj` there)
+    public static Field resolveSubscriptOnAliasedField(ColumnIdent path,
+                                                       Fields fields,
```

Comment:
based on the signature it's not clear if `fields` and `fieldResolver` are from the child relation or from the outer relation. Maybe add some docs?